### ISSUE TITLE
[Feature request] - Load weights locally on applications

### ISSUE THREAD

**Issue by maifeeulasad** [keras-team/keras#14321]
I'm working on a project, that is being designed to work on offline. So I downloaded the pre-trained VGG16 weights and tried passing it as a parameter, but it doesn't work as the code says it shouldn't.
```
from tensorflow.keras.applications.vgg16 import VGG16

model = VGG16(
            weights = "../input/model_weights/vgg16_weights_tf_dim_ordering_tf_kernels.h5",
            ...
)
```

By any chance can this feature be included in future Keras ?

And recreating the model causes tons of errors, such as : 
 - https://stackoverflow.com/q/65389153/10305444
 - https://stackoverflow.com/q/49113140/10305444

Line responsible for these : https://github.com/keras-team/keras/blob/1d49681f825d2d4282719efd27e13aca441b39c2/keras/applications/vgg16.py#L209_L222

>>>The error it produces : 
**You are trying to load a weight file containing 16 layers into a model with 13 layers.**

**Comment by patel-rupin2000** [comment]
model=Sequential()
from tensorflow.keras.applications.vgg16 import VGG16
base_model =  VGG16(input_shape = (224, 224, 3), include_top = False, weights = 'imagenet')
for layer in base_model.layers:
    layer.trainable = False
x = Flatten()(base_model.output)
x = Dense(1024, activation='relu')(x)
x = Dropout(0.5)(x)
x = Dense(3, activation='softmax')(x)
model = Model(base_model.input, x)
model.compile(optimizer = "adam", loss = 'categorical_crossentropy',metrics = ["accuracy"])
model.summary()

Model: "model_1"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
input_2 (InputLayer)         [(None, 224, 224, 3)]     0         
_________________________________________________________________
block1_conv1 (Conv2D)        (None, 224, 224, 64)      1792      
_________________________________________________________________
block1_conv2 (Conv2D)        (None, 224, 224, 64)      36928     
_________________________________________________________________
block1_pool (MaxPooling2D)   (None, 112, 112, 64)      0         
_________________________________________________________________
block2_conv1 (Conv2D)        (None, 112, 112, 128)     73856     
_________________________________________________________________
block2_conv2 (Conv2D)        (None, 112, 112, 128)     147584    
_________________________________________________________________
block2_pool (MaxPooling2D)   (None, 56, 56, 128)       0         
_________________________________________________________________
block3_conv1 (Conv2D)        (None, 56, 56, 256)       295168    
_________________________________________________________________
block3_conv2 (Conv2D)        (None, 56, 56, 256)       590080    
_________________________________________________________________
block3_conv3 (Conv2D)        (None, 56, 56, 256)       590080    
_________________________________________________________________
block3_pool (MaxPooling2D)   (None, 28, 28, 256)       0

**Comment by jvishnuvardhan** [comment]
@maifeeulasad Sorry for the late response. Is this still an issue for you? Can you please share a simple standalone code to reproduce the issue/feature you are looking? Thanks

@patel-rupin2000 I ran your code without any issue. [Here](https://colab.research.google.com/gist/jvishnuvardhan/a474dab153d2a437d768afa4cb1634c5/untitled1177.ipynb) is a gist for reference. Thanks!
